File: Backend/app/models/transport_request.py
from dataclasses import dataclass
from datetime import datetime
from uuid import UUID

from models.location import Location
from models.priority import Priority
from models.status import Status


@dataclass
class TransportRequest:
    id: UUID
    pickup_location: Location
    drop_off_location: Location
    assigned_timeframe: datetime
    priority: Priority
    status: Status
    assigned_service: str

File: frontend/src/types/TransportRequest.ts
export type LocationType = "STATION" | "PATIENT_ROOM" | "OP" | "EMERGENCY_ROOM";
export type PriorityType = 0 | 1 | 2 | 3;
export type StatusType = "OPEN" | "ASSIGNED" | "IN_PROGRESS" | "CLOSED" | "CANCELLED";

export interface TransportRequest {
    id: string;
    status: StatusType;
    pickup_location: LocationType;
    drop_off_location: LocationType;
    assigned_timeframe: string;
    priority: PriorityType;
}
